examples/pennylane_universal_classifier.py

```diff
@@ -0,0 +1,474 @@
+make a universal classifier (Cybenko 1989). Here is where things gets a
+bit quantum. After the application of the layers, we will end up at some
+point on the Bloch sphere due to the sequence of unitaries implementing
+rotations of the input. These are still just linear transformations of
+the input state. Now, the output of the model should be a class label
+which can be encoded as fixed vectors (Blue = :math:`[1, 0]`, Red =
+:math:`[0, 1]`) on the Bloch sphere. We want to end up at either of them
+after transforming our input state through alternate applications of
+data layer and weights.
+
+We can use the idea of the “collapse” of our quantum state into
+one or other class. This happens when we measure the quantum state which
+leads to its projection as either the state 0 or 1. We can compute the
+fidelity (or closeness) of the output state to the class label making
+the output state jump to either :math:`| 0 \rangle` or
+:math:`|1\rangle`. By repeating this process several times, we can
+compute the probability or overlap of our output to both labels and
+assign a class based on the label our output has a higher overlap. This
+is much like having a set of ouput neurons and selecting the one which
+has the highest value as the label.
+
+In Pennylane, we can define an observable (the expected output
+label) and make a circuit to return the fidelity using the
+`Hermitian <https://pennylane.readthedocs.io/en/latest/code/ops/qubit.html#pennylane.ops.qubit.Hermitian>`_
+operator. We can then define the cost function as the sum of the
+fidelities for all the data points after passing throuh the circuit
+and optimize the parameters :math:`(\vec \theta)` to minimize the cost.
+
+.. math::
+
+
+   \texttt{Cost} = \sum_{\texttt{data points}} (1 - \texttt{fidelity}(\psi_{\texttt{output}}(\vec x, \vec \theta), \psi_{\texttt{label}}))
+
+Now, we can use our favorite optimizer to maximize the sum of the
+fidelities over all data points (or batches of datapoints) and find the
+optimal weights for classification. Gradient based optimizers such as
+Adam can be used if we have a good model of the circuit and how noise
+might affect it. Or, we can use some gradient-free method such as LBFGS
+to evaluate the gradient and find the optimal weights where we can treat
+the quantum circuit as a black-box and the gradients are computed
+numerically using a fixed number of function evalutaions and iterations.
+
+Multiple qubits, entanglement and Deep Neural Networks
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The Universal Approximation Theorem declares a single hidden layered
+neural network to be capable of approximating any function to arbitrary
+accuracy. But in practice, it might require a large number of neurons in
+the single hidden layer and here is where Deep Neural Networks come into
+action. Deep Neural Networks proved to be better in practice and we have
+some intuitive idea why, read “Why does deep and cheap learning work so
```

Comment:
I'd be careful, there are lots of explanations, especially from the last two years that are not mentioned in that paper...